### PR TITLE
feat: added jabba

### PR DESCRIPTION
**Pull request by PavelVesnin**

### CODE REVIEW

File: apps/graphql/test/web/resolvers/legal_entity_deactivation_job_resolver_test.exs [hunk 1]
```diff
@@ -348,18 +276,23 @@ defmodule GraphQL.LegalEntityDeactivationJobResolverTest do
...
       variables = %{id: id}
 
-      resp =
+      resp_body =
         conn
         |> post_query(query, variables)
         |> json_response(200)
-        |> get_in(~w(data legalEntityDeactivationJob))
+
+      assert nil == resp_body["errors"]
```
Comment (progsmile) [inline apps/graphql/test/web/resolvers/legal_entity_deactivation_job_resolver_test.exs]:
```suggestion
      refute resp_body["errors"]
```

File: apps/graphql/test/web/resolvers/legal_entity_deactivation_job_resolver_test.exs [hunk 2]
```diff
@@ -381,8 +314,31 @@ defmodule GraphQL.LegalEntityDeactivationJobResolverTest do
 
   defp create_job(legal_entity) do
     meta = %{deactivated_legal_entity: Map.take(legal_entity, ~w(id name edrpou)a)}
-    {:ok, job} = TasKafkaJobs.create(meta, @legal_entity_deactivation_type)
-    {:ok, ObjectId.encode!(job._id), job}
+    #    {:ok, job} = TasKafkaJobs.create(meta, @legal_entity_deactivation_type)
```
Comment (progsmile) [inline apps/graphql/test/web/resolvers/legal_entity_deactivation_job_resolver_test.exs]:
```suggestion
```

File: apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs [hunk 3]
```diff
@@ -52,10 +50,21 @@ defmodule GraphQL.LegalEntityMergeJobResolverTest do
       from = insert(:prm, :legal_entity)
       to = insert(:prm, :legal_entity)
       insert(:prm, :related_legal_entity, merged_to: to)
+      job = job("merge_legal_entities", "PENDING", to, from)
+
+      expect(RPCWorkerMock, :run, fn _, _, :create_job, [tasks, _type, _opts] ->
+        assert 1 == length(tasks)
+        %{name: name, callback: {_, m, f, a}} = hd(tasks)
```
Comment (progsmile) [inline apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs]:
```suggestion
        %{name: name, callback: {_, module, function, args}} = hd(tasks)
```

File: apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs [hunk 4]
```diff
@@ -502,25 +439,27 @@ defmodule GraphQL.LegalEntityMergeJobResolverTest do
     end
   end
 
-  defp create_job(merged_from, merged_to) do
+  defp job(type, status, merged_to, merged_from, ended_at \\ nil) do
```
Comment (progsmile) [inline apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs]: how about job_factory?

File: apps/jobs/lib/jabba/client.ex [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+    job = Map.put(job, :meta, meta)
+
+    # ToDo: use inserted_at instead
+    job = Map.put(job, :started_at, job.inserted_at)
+
+    # ToDO: do not encode to JSON. Talk about that with frontend team
+    case Jason.encode(job.result) do
+      {:ok, json} -> Map.put(job, :result, json)
+      _ -> Map.put(job, :result, inspect(job.result))
```
Comment (progsmile) [inline apps/jobs/lib/jabba/client.ex]: That's incorrect for sure, should respond with the same format

File: apps/jobs/lib/jobs/legal_entity_deactivation_job.ex [hunk 6]
```diff
@@ -18,20 +17,37 @@ defmodule Jobs.LegalEntityDeactivationJob do
...
+  def search_jobs(filter, order_by, limit, offset) do
+    filter
+    |> Keyword.put(:type, @legal_entity_deactivation_type)
+    |> JabbaClient.search_jobs(order_by, {limit, offset})
+  end
+
+  def get_job(id) do
+    case JabbaClient.get_job(id) do
+      {:ok, job} -> {:ok, prepare_meta(job)}
+      nil -> {:ok, nil}
```
Comment (progsmile) [inline apps/jobs/lib/jobs/legal_entity_deactivation_job.ex]: Why result is `{:ok, nil}` ? It's not ok that value not found

File: apps/jobs/lib/jobs/legal_entity_deactivation_job.ex [hunk 7]
```diff
@@ -42,79 +58,66 @@ defmodule Jobs.LegalEntityDeactivationJob do
...
+        {:error, errors}
 
       {:error, reason} ->
-        Logger.error("failed to deactivate legal entity with: #{inspect(reason)}")
-        TasKafkaJobs.failed(job_id, reason)
+        Logger.error("Failed to deactivate legal entity with: #{inspect(reason)}")
         {:error, reason}
     end
   rescue
     e ->
```
Comment (progsmile) [inline apps/jobs/lib/jobs/legal_entity_deactivation_job.ex]:
```suggestion
    err ->
```

File: apps/jobs/lib/jobs/legal_entity_deactivation_job.ex [hunk 8]
```diff
@@ -42,79 +58,66 @@ defmodule Jobs.LegalEntityDeactivationJob do
...
 
     with {:ok, legal_entity} <- LegalEntities.fetch_by_id(id),
          :ok <- check_transition(legal_entity),
          :ok <- LegalEntities.check_nhs_reviewed(legal_entity, true) do
-      job_data = get_legal_entity_deactivation_event_data(legal_entity, user_id)
-      meta = %{deactivated_legal_entity: Map.take(legal_entity, ~w(id name edrpou)a)}
+      tasks = get_legal_entity_deactivation_tasks(legal_entity, user_id)
+
+      opts = [
+        name: "Deactivate legal entity",
```
Comment (progsmile) [inline apps/jobs/lib/jobs/legal_entity_deactivation_job.ex]: Does this name serve only for description purposes or it's used within functions?

File: apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs (hunk 3 above)
Comment (PavelVesnin) [reply apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs]: I believe that `m, f, a` is clearly to understand what is it

File: apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs (hunk 4 above)
Comment (PavelVesnin) [reply apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs]: there is no DB for job. Do you think that this ok, when you can `build` but can't `insert` struct?

File: apps/jobs/lib/jabba/client.ex (hunk 5 above)
Comment (PavelVesnin) [reply apps/jobs/lib/jabba/client.ex]: This is the most uncomfortable place in code. Result of callback should be formatted in correct way and validated on Jabba. I didn't find yet optimal way to implement this.
In next development iterations in Jabba this case should be resolved

File: apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs (hunk 4 above)
Comment (AlexKovalevych) [reply apps/graphql/test/web/resolvers/legal_entity_merge_job_resolver_test.exs]: absolutely

**Pull request merged**